libpimeval/src/pimSim.cpp

```diff
@@ -468,7 +468,8 @@ pimSim::pimBroadcast(PimObjId dest, T value)
 {
   pimPerfMon perfMon("pimBroadcast");
   if (!isValidDevice()) { return false; }
-  std::unique_ptr<pimCmd> cmd = std::make_unique<pimCmdBroadcast<T>>(PimCmdEnum::BROADCAST, dest, value);
+  uint64_t signExtBits = pimUtils::castTypeToBits(value);
```

Comment:
This is a key change that handles template type here, so that pimCmdBroadcast does not need to be a template class